server/server.go

```diff
@@ -232,29 +238,28 @@ func (s *server) Ping(context.Context, *rpcpb.PingRequest) (*rpcpb.PingResponse,
 	return &rpcpb.PingResponse{Pid: int32(os.Getpid())}, nil
 }
 
-const defaultStartTimeout = 5 * time.Minute
+func (s *server) Start(_ context.Context, req *rpcpb.StartRequest) (*rpcpb.StartResponse, error) {
+	s.mu.Lock()
```

Comment:
this is a lock over server network/clusterinfo data, another PR was trying to reduce the space of server mutual exclusion by reducing the lock only to parts of code that modify them. This PR is expanding the lock even more

Reply:
> This PR is expanding the lock even more

I think this is what it takes in order for `server` to be threadsafe -- we can't have multiple goroutines simultaneously accessing the state of `server` or else it's unsafe